### DISCUSSION TITLE
Get active keymap

### DISCUSSION THREAD

**Discussion by mshzsh** [hyprwm/Hyprland#1905] [Q&A] [answered]
How can I find out what active keymap is?
I used  `hyprctl devices` command  before, but when I connect an external keyboard, this method no longer works!

**Comment by vaxerski** [comment, marked as answer]
wdym no longer works

**Reply by mshzsh** [reply]
solution? to find out which is the active keyboard

**Reply by vaxerski** [reply]
what?

active keyboard has `main: yes` in `hyprctl devices`

I don't understand what is not working here

**Reply by mshzsh** [reply]
I use several different languages to write on the keyboard, I want to know if my keyboard language is English or German or Persian or Chinese or...
How should I understand?

**Reply by vaxerski** [reply]
![image](https://user-images.githubusercontent.com/43317083/228313091-b89d3e07-0874-4d30-8f6b-5047948ae4c3.png)

Active keymap: Polish

??

**Reply by mshzsh** [reply]
for example if you use a laptop keyboard, you should see `dell-kb216-wired-keyboard`, now if you connect an external keyboard, the `dell-kb216-wired-keyboard` value will not change and you should see the `at-translated-set-2-keyboard` value! I want it to be something fixed so that I don't change the codes

**Reply by vaxerski** [reply]
use the `-j` flag with `jq` and pick up the value of the one that has `main` set to `true`

**Reply by mshzsh** [reply]
thanks!

**Reply by vaxerski** [reply]
keep in mind there may be two `main` ones if you have a virtual keyboard as well (e.g. IME, VNC, etc)

**Reply by mshzsh** [reply]
Yes, and right now there is this problem that if I change the language of the laptop keyboard, the language of the external keyboard does not change and vice versa.

**Reply by vaxerski** [reply]
if you want that, you'll need to make a keybind in hyprland and just execute two separate `hyprctl switchxkblayout` commands for the two keyboards.

**Reply by mshzsh** [reply]
Great :+1:

**Reply by mshzsh** [reply]
Is it possible to use `Alt` and `Shift` combination keys?
Because it doesn't work : 
`bind=ALT,SHIFT,exec,...`
and
`bind=,ALT_SHIFT,exec,...`

**Reply by vaxerski** [reply]
https://wiki.hyprland.org/Configuring/Binds/#binding-mods

**Reply by mshzsh** [reply]
It worked with `bind=ALT,SHIFT_L,exec`, but after releasing the keys, `ALT` seems to work again and there is a program menu on the screen, and it gets focused, while if I put any key except `ALT + SHIFT`, this does not happen. Is there a solution for this or should I choose another key?
